### PR TITLE
[WIP] Use WebSockets for uploading/downloading files and directories (using callbacks).

### LINKED ISSUE #929: Support bulk load in `neuro cp`
See https://github.com/neuromation/platform-storage-api/issues/114

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
Closes #929.

### CODE REVIEW

File: neuromation/api/storage.py [hunk 1]
```diff
@@ -279,24 +359,38 @@ def _uri_to_path(self, uri: URL) -> str:
...
-                        errno.ENOTDIR, "Not a directory", str(dst.parent)
-                    )
-            except ResourceNotFound:
-                raise NotADirectoryError(
-                    errno.ENOTDIR, "Not a directory", str(dst.parent)
+
+        async with self._ws_connect(dst, "WEBSOCKET_WRITE") as ws:
+            await self._upload_file(ws, src, path, dst, "", progress=progress)
+
+    # XXX Move to WSStorageClient?
```
Comment (asvetlov) [inline neuromation/api/storage.py]: Up to you

**Comment by serhiy-storchaka** [comment]
Tests are failed because the API is tested with a small builtin storage server, which just does not support WebSocket operations yet.

**Comment by serhiy-storchaka** [comment]
Some benchmarking results.

Master:
39-45 seconds for `neuro storage cp -r neuromation storage:`
32-45 seconds for `neuro storage cp -r storage:neuromation /tmp`
7 seconds for `neuro storage cp neuromation.tar storage:`
2 seconds for `neuro storage cp storage:neuromation.tar /tmp`

Current PR:
7-8 seconds for `neuro storage cp -r neuromation storage:`
3-4 seconds for `neuro storage cp -r storage:neuromation /tmp`
8 seconds for `neuro storage cp neuromation.tar storage:`
3 seconds for `neuro storage cp storage:neuromation.tar /tmp`

See also #958.

**Comment by asvetlov** [comment]
Super impressive numbers!

**Comment by asvetlov** [comment]
Aha, I see `if not self._handlers: break` in `run()`.

**Pull request closed without merging**